src/query/storages/fuse/src/operations/merge_into/processors/processor_merge_into_matched_and_split.rs

```diff
@@ -318,8 +319,44 @@ impl Processor for MatchedSplitProcessor {
                 current_block = op.execute(&self.ctx.get_function_context()?, current_block)?;
                 metrics_inc_merge_into_append_blocks_counter(1);
                 metrics_inc_merge_into_append_blocks_rows_counter(current_block.num_rows() as u32);
+
+                // cornor case: for merge into update, if the target table's column is not null,
```

Comment:
Prefer spill these codes to a new function, like:
```
normalize_column_types_for_merge(ctx, ...) -> Result<Datablock>
```
To make the process logic more clear and more modular.

Reply:
ok